src/test/java/StringManipulationTest.java

```diff
@@ -87,63 +136,101 @@ public void testGeSubStrings1() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedstring.getSubStrings(1, 0);});
     }
+
+    // Test when the startWord index is greater than endWord index
+    // Throws IllegalArgumentException
     @Test
     public void testGeSubStrings4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is my string");
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedstring.getSubStrings(2, 1);});
     }
+
+    // Test when the endWord index is greater than the word count of string
+    // Throws IndexOutOfBoundException
     @Test
     public void testGeSubStrings5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is my string");
+        assertThrows(IndexOutOfBoundsException.class, () -> {
+            manipulatedstring.getSubStrings(2, 5);});
     }
+
+    // Test when the startWord and endWord are equal
     @Test
     public void testGeSubStrings6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is my string");
+        String [] sStings = manipulatedstring.getSubStrings(4, 4);
+        String expected = "string";
+        String actual = sStings[0];
+
+        assertEquals(expected, actual);
     }
 
     @Test
     public void testRestoreString1()
     {
         manipulatedstring.setString("art");
         int [] array;
-        array=new int[]{1,0,2};
+        array = new int[]{1,0,2};
         String restoreString = manipulatedstring.restoreString(array);
         assertEquals(restoreString, "rat");
     }
 
+    // Throw IllegalArgumentException if the indices length is greater than the string length. {5,6,7,8,9,10,11,0,1,2,3,4,12};
     @Test
     public void testRestoreString2()
     {
-        fail("Not yet implemented");
-
+        manipulatedstring.setString("JUnitTesting!");
+        int[] indices = {5,6,7,8,9,10,11};
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedstring.restoreString(indices);});
     }
 
+    // Throw IndexOutOfBoundsException if any indice is less than 0.
     @Test
     public void testRestoreString3()
     {
-        fail("Not yet implemented");
-
+        manipulatedstring.setString("JUnitTesting!");
+        int[] indices = {5,6,7,8,9,10,11,0,1,2,3,4,-8};
+        assertThrows(IndexOutOfBoundsException.class, () -> {
+            manipulatedstring.restoreString(indices);});
     }
 
+    // Throw IndexOutOfBoundsException if any indice is greater than the string length
     @Test
     public void testRestoreString4()
     {
-        fail("Not yet implemented");
-
+        manipulatedstring.setString("JUnitTesting!");
+        int[] indices = {5,6,7,8,9,10,11,0,1,2,3,4,13};
+        assertThrows(IndexOutOfBoundsException.class, () -> {
+            manipulatedstring.restoreString(indices);});
     }
 
+    // Test when each character is the same in the string
```

Comment:
This is an interesting test case, I wonder in what ways it would break/why it would be needed

Reply:
Yes, it is an interesting test case. I am not sure how you would break this case too. I run out of ideas and made up a case lol. Maybe it could be useful when implementation takes shortcuts or makes assumptions if the characters are the same. This test case ensures that the code does not make invalid assumptions.